src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/shim.cxx

```diff
@@ -1052,10 +1075,19 @@ namespace xclhwemhal2 {
     {
       mEnvironmentNameValueMap["enable_pr"] = "false";
     }
-
+    //Providing enough time to simulate script to set up it's environment and start xsim.
+    using namespace std::chrono_literals;
+    std::this_thread::sleep_for(10s);
+    if (parseLog() != 0) {
+      if (mLogStream.is_open())
+        mLogStream << __func__ << " ERROR: [HW-EMU 26] Simulator is NOT started so exiting the application! " << std::endl;
+      // If no simulator running then no need to try a connection, hence exit with a failure now.
+      exit(EXIT_FAILURE);
```

Comment:
I am really concerned about all these exits,

Reply:
It is replaced with a run_time exception which ensures the program is terminated with clear error on console.

Reply:
I don't think you should exit at all.